### PR TITLE
br: add integration test for ingest corner case test

### LINKED ISSUE #52733: when creating index, log backup should be able to start
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. add index
2. start log backup

### 2. What did you expect to see? (Required)
done
### 3. What did you see instead (Required)
```
Unable to create log backup task. Please wait until the DDL jobs(add index with ingest method) are finished.
```
### 4. What is your TiDB version? (Required)
7.1, 7.5, 8.1, master

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #52733

Problem Summary:
When creating index, log backup should be able to start.
### What changed and how does it work?
remove the check, and add integration test.
### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/tests/br_pitr_failpoint/check/check_ingest_repair.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+set -eu
+
+# check index schema
+## check table test.pairs
+run_sql "SHOW INDEX FROM test.pairs WHERE Key_name = 'i1';"
+check_contains "Column_name: y"
+check_contains "Column_name: z"
+
+# check index data
```
Comment (lance6716) [inline br/tests/br_pitr_failpoint/check/check_ingest_repair.sh:25]: also ADMIN CHECK TABLE to find data-index inconsistency

File: br/tests/br_pitr_failpoint/run.sh [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+export GO_FAILPOINTS="github.com/pingcap/tidb/pkg/ddl/create-index-stuck-before-public=1*return(\"$hint_sig_file_public\");\
+github.com/pingcap/tidb/pkg/ddl/create-index-stuck-before-ddlhistory=1*return(\"$hint_sig_file_history\")"
+
+# prepare the intersect data
+run_sql_file $CUR/intersect_data/ingest_repair1.sql &
+sql_pid=$!
+
+# start the log backup task
+echo "start log task"
+run_br --pd $PD_ADDR log start --task-name integration_test -s "local://$TEST_DIR/$PREFIX/log"
```
Comment (lance6716) [inline br/tests/br_pitr_failpoint/run.sh:45]: br should wait until DDL really started 🤔

**Review by lance6716 — APPROVED** [review]

File: br/tests/br_pitr_failpoint/check/check_ingest_repair.sh (hunk 1 above)
Comment (Leavrth) [reply br/tests/br_pitr_failpoint/check/check_ingest_repair.sh:25]: done

File: br/tests/br_pitr_failpoint/run.sh (hunk 2 above)
Comment (Leavrth) [reply br/tests/br_pitr_failpoint/run.sh:45]: In the shell function `start_tidb`, it checks TiDB by accessing `https://$TIDB_IP:10080/status`.

Comment (lance6716) [reply br/tests/br_pitr_failpoint/run.sh:45]: Sorry I mean wait until ADD INDEX is running

Comment (Leavrth) [reply br/tests/br_pitr_failpoint/run.sh:45]: we just need the log data from [full-backup-ts, +inf).

Comment (Leavrth) [reply br/tests/br_pitr_failpoint/run.sh:45]: Get

**Review by BornChanger — APPROVED** [review]

File: pkg/ddl/index.go [hunk 3]
```diff
@@ -862,32 +878,6 @@ func cleanupSortPath(ctx context.Context, currentJobID int64) error {
 	return nil
 }
 
-// IngestJobsNotExisted checks the ddl about `add index` with ingest method not existed.
```
Comment (Benjamin2037) [inline pkg/ddl/index.go:865]: Why here deleted this func?

Comment (BornChanger) [reply pkg/ddl/index.go:865]: This func is used when log backup is not compatible with fast create index. In 7.0+, it's useless.

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**